scripts/staging/fm-binclass.dml

```diff
@@ -0,0 +1,188 @@
+          loss_reg_w0 = l2_reg::forward(w0, lambda);
+          loss_reg_W  = l2_reg::forward(W, lambda);
+          loss_reg_V  = l2_reg::forward(V, lambda);
+          
+          accuracy = mean((loss<0.5) == (y_batch<0.5));
+          loss = loss_data + loss_reg_X + loss_reg_w0 + loss_reg_W + loss_reg_V;
+          
+          # Compute validation loss & accuracy
+          probs_val = predict(X_val, w0, W, V)
+          #loss_val = cross_entropy_loss::forward(probs_val, y_val)
+          loss_val = log_loss::forward(probs_val, y_val)
+          accuracy_val = mean((probs_val<0.5) == (y_val<0.5))
+
+          # Output results
+          print("Epoch: " + e + ", Iter: " + i + ", Train Loss: " + loss + ", Train Accuracy: "
+                + accuracy + ", Val Loss: " + loss_val + ", Val Accuracy: " + accuracy_val)
+        }
+        
+        # 6.Send the result of sigmoid::forward and the correct labels y to log_loss::backward
+        dsfy = log_loss::backward(sfy, y_batch);
+    
+        # 7.Send the above result through sigmoid::backward
+        dy = sigmoid::backward(dsfy, y_res);
+    
+        # 8.Send the above result through fm::backward
+        [dw0, dW, dV] = fm::backward(dy, X_batch, w0, W, V);
+        
+        # 9.Call adam::update for all parameters
+
+        #X = adam::update(X, dX, lr, beta1, beta2, epsilon, t, mX, vX);
+        [w0, mw0, vw0] = adam::update(w0, dw0, lr, beta1, beta2, epsilon, t, mw0, vw0);
+        [W, mW, vW]  = adam::update(W, dW, lr, beta1, beta2, epsilon, t, mW, vW );
+        [V, mV, vV]  = adam::update(V, dV, lr, beta1, beta2, epsilon, t, mV, vV );
+      }
+    }
+}
+
+predict = function(matrix[double] X, matrix[double] w0, matrix[double] W, matrix[double] V)
+    return (matrix[double] out) {
+  /*
+   * Computes the predictions for the given inputs.
+   *
+   * Inputs:
+   *  - X : n examples with d features, of shape (n, d).
+   *  - w0, W, V : trained model parameters. 
+   *  
+   * Outputs:
+   *  - out : target vector, y.
+   */
+    
+    # 1.initialize fm core
+    #[w0, W, V] = fm::init(d, k);
+    
+    # 2.Send inputs through fm::forward
+    y = fm::forward(X, w0, W, V);
+    
+    # 3.Send the above result through sigmoid::forward
+    out = sigmoid::forward(y);
+    
+    # 4.Send the above result through log_loss::forward
+    # loss = log_loss::forward(out); 
+  
+}
+
+eval = function(matrix[double] probs, matrix[double] y)
+    return (double loss, double accuracy) {
+
+    # Computes loss and accuracy.
+    # 1. just compute loss with log_loss::forward(probs, y)
+    loss = log_loss::forward(probs, y);
+
+    # 2. compute accuracy
+    accuracy = mean( (probs<0.5) == (y<0.5) )
+}
+
+# please delete this after before committing !
+X = rand(rows=1000, cols=4); n = 1000; d = 4; k=2;
+y = rand(rows=n, cols=1);
+X_val = rand(rows=100, cols=4);
+y_val = rand(rows=100, cols=1);
+
+[w0, W, V, loss] = train(X, y, X_val, y_val);
+print(as.scalar(w0));
```

Comment:
I would place all this into a `fm-binclass-train-dummy.dml` script, similar to [`https://github.com/apache/systemml/blob/master/scripts/nn/examples/mnist_lenet_distrib_sgd-train-dummy-data.dml`](https://github.com/apache/systemml/blob/master/scripts/nn/examples/mnist_lenet_distrib_sgd-train-dummy-data.dml) (ignore the distributed stuff).